### ISSUE TITLE
Cannot Get Redis Key Value

### ISSUE THREAD

**Issue by btahir** [upstash/upstash-redis#277]
Hi - 

Not sure what I am doing wrong but I cannot seem to get the Redis key value.

If I do something like:

```
  await redis.set('key', 'value',)
  const data = await redis.get('key')
  console.log(data)
```

I get back a hashed string like `dmFsdWU=`. How do I resolve this to the actual value ('value'). I am using @upstash/redis in Nextjs api routes.

**Comment by chronark** [comment]
Please try creating your redis instance like this:
```ts
new Redis({...., responseEncoding:false })
```

I'm investigating this problem right now

can you tell me your redis version and runtime?
also when did this start happening?

**Comment by chronark** [comment]
I can't reproduce this, could you maybe create a repo where this happens?

I tried:
```bash
npx create-next-app nextjs
cd nextjs
pnpm add @upstash/redis@latest
```

Then updated `/api/hello.ts`
```ts
// Next.js API route support: https://nextjs.org/docs/api-routes/introduction
import { Redis } from "@upstash/redis"
export default async function handler(req, res) {
  const redis = Redis.fromEnv()

  await redis.set('key', 'value',)
  const data = await redis.get('key')
  console.log(data)
  res.status(200).json({ data })
}

```
Then deployed to vercel
```bash
npx vercel
```

when I visit `/api/hello` it returns `{"data":"value"}` as expected

**Comment by btahir** [comment]
> Please try creating your redis instance like this:
> 
> ```ts
> new Redis({...., responseEncoding:false })
> ```
> 
> I'm investigating this problem right now
> 
> can you tell me your redis version and runtime? also when did this start happening?

This seemed to fix it! I literally spun this up last night (and deleted and created another db today for good measure) so its the latest versions. You can close issue or keep it open until a more thorough resolution up to you.

Thanks!

**Comment by chronark** [comment]
sorry I meant our `@upstash/redis` version
and what runtime version are you using in vercel?
node14,16 or 18?

**Comment by btahir** [comment]
> and what runtime version are you using in vercel? node14,16 or 18?

Node 14 with @upstash/redis 1.18.1.

**Comment by anishsrinivasan** [comment]
Yes, even I faced the same issue when it was deployed on Vercel. 

`responseEncoding: false`, seems to be the solution for now. 
```tsx
new Redis({...., responseEncoding:false })
```

I am assuming, this could be because of the NodeJS version (Node 14), whereas it's working fine with Node (v18.12.0).

**Comment by chronark** [comment]
Yes I just figured this out as well.

I'll be working on a fix for nodejs14 but in the meantime you need to either use `responseEncoding: false` or upgrade to at least node v16

**Comment by chronark** [comment]
Disregard the comment above, I fixed it in `v1.18.2`
